### PR TITLE
[GCC 8] [RECO] Fixed const correctness for operator()

### PR DESCRIPTION
**Pull request by smuzaffar**

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: DataFormats/METReco/interface/HcalNoiseRBX.h [hunk 1]
```diff
@@ -97,7 +97,7 @@ namespace reco {
     
     // helper function to get the unique calotowers
     struct twrcomp {
-      inline bool operator() ( const CaloTower & t1, const CaloTower & t2 ) {
+      inline bool operator() ( const CaloTower & t1, const CaloTower & t2 ) const{
```
Comment (Dr15Jones) [inline DataFormats/METReco/interface/HcalNoiseRBX.h]: This overlaps with #23235

File: RecoVertex/MultiVertexFit/interface/LinTrackCache.h [hunk 2]
```diff
@@ -12,12 +12,12 @@ class LinTrackCache
 private:
   struct Comparer
   {
-    bool operator() ( const GlobalPoint &, const GlobalPoint & );
```
Comment (Dr15Jones) [inline RecoVertex/MultiVertexFit/interface/LinTrackCache.h]: This overlaps with #23237

**Comment by perrotta** [comment]
please test
(to get rid of the effect of the extra commits on jenkins outputs)

**Comment by slava77** [comment]
-1

need to decouple from #23237 and #23235

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by smuzaffar** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass with no differences

**Comment by fabiocos** [comment]
+1

**Pull request merged**